cirq/google/engine/calibration.py

```diff
@@ -103,8 +113,48 @@ def __len__(self) -> int:
         return len(self._metric_dict)
 
     def __str__(self) -> str:
-
-        return 'Calibration(keys={})'.format(list(sorted(self.keys())))
+        return f'Calibration(keys={list(sorted(self.keys()))})'
+
+    def __repr__(self) -> str:
+        return ('cirq.google.Calibration(metrics='
+                f'{repr(dict(self._metric_dict))})')
+
+    def to_proto(self) -> v2.metrics_pb2.MetricsSnapshot:
+        """Reconstruct the protobuf message represented by this class."""
+        proto = v2.metrics_pb2.MetricsSnapshot()
+        for key in self._metric_dict:
+            for target, value_list in self._metric_dict[key].items():
+                current_metric = proto.metrics.add()
+                current_metric.name = key
+                current_metric.targets.extend(
+                    [v2.qubit_to_proto_id(q) for q in target])
+                for value in value_list:
+                    current_value = current_metric.values.add()
+                    if isinstance(value, float):
+                        current_value.double_val = value
+                    elif isinstance(value, int):
+                        current_value.int64_val = value
+                    elif isinstance(value, str):
+                        current_value.str_val = value
+                    else:
+                        raise ValueError(f'Unsupported metric value {value}. '
+                                         'Must be int, float, or str to '
+                                         'convert to proto.')
+        return proto
+
+    @classmethod
+    def _from_json_dict_(cls, metrics: str, **kwargs):
+        """Magic method for the JSON serialization protocol."""
+        metric_proto = v2.metrics_pb2.MetricsSnapshot()
+        json_format.Parse(metrics, metric_proto)
+        return cls(metric_proto)
+
+    def _json_dict_(self):
+        """Magic method for the JSON serialization protocol."""
+        return {
+            'cirq_type': 'Calibration',
+            'metrics': json_format.MessageToJson(self.to_proto())
```

Comment:
Can use `json_format.MessageToDict` to convert to a dict that can then be converted to string by the standard json serialization. This avoids the json-string-in-json problem. Similarly, use `json_format.ParseDict` instead of `json_format.Parse` when deserializing.

Reply:
Ooohhh, that's just what I was looking for.  Let me try that out!